bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapOneCallHandler.java

```diff
@@ -0,0 +1,636 @@
+                if (forecastHours > newForecastHours) {
+                    for (int i = newForecastHours + 1; i <= forecastHours; i++) {
+                        toBeRemovedChannels.addAll(removeChannelsOfGroup(
+                                CHANNEL_GROUP_HOURLY_FORECAST_PREFIX + ((i < 10) ? "0" : "") + Integer.toString(i)));
+                    }
+                } else {
+                    for (int i = forecastHours + 1; i <= newForecastHours; i++) {
+                        toBeAddedChannels.addAll(createChannelsForGroup(
+                                CHANNEL_GROUP_HOURLY_FORECAST_PREFIX + ((i < 10) ? "0" : "") + Integer.toString(i),
+                                CHANNEL_GROUP_TYPE_ONECALL_HOURLY_FORECAST));
+                    }
+                }
+                forecastHours = newForecastHours;
+            }
+            if (forecastDays != newForecastDays) {
+                logger.debug("Rebuilding daily forecast channel groups.");
+                if (forecastDays > newForecastDays) {
+                    if (newForecastDays < 1) {
+                        toBeRemovedChannels.addAll(removeChannelsOfGroup(CHANNEL_GROUP_FORECAST_TODAY));
+                    }
+                    if (newForecastDays < 2) {
+                        toBeRemovedChannels.addAll(removeChannelsOfGroup(CHANNEL_GROUP_FORECAST_TOMORROW));
+                    }
+                    for (int i = newForecastDays; i < forecastDays; ++i) {
+                        toBeRemovedChannels.addAll(
+                                removeChannelsOfGroup(CHANNEL_GROUP_DAILY_FORECAST_PREFIX + Integer.toString(i)));
+                    }
+                } else {
+                    if (forecastDays == 0 && newForecastDays > 0) {
+                        toBeAddedChannels.addAll(createChannelsForGroup(CHANNEL_GROUP_FORECAST_TODAY,
+                                CHANNEL_GROUP_TYPE_ONECALL_DAILY_FORECAST));
+                    }
+                    if (forecastDays <= 1 && newForecastDays > 1) {
+                        toBeAddedChannels.addAll(createChannelsForGroup(CHANNEL_GROUP_FORECAST_TOMORROW,
+                                CHANNEL_GROUP_TYPE_ONECALL_DAILY_FORECAST));
+                    }
+                    for (int i = (forecastDays < 2) ? 2 : forecastDays; i < newForecastDays; ++i) {
+                        toBeAddedChannels.addAll(
+                                createChannelsForGroup(CHANNEL_GROUP_DAILY_FORECAST_PREFIX + Integer.toString(i),
+                                        CHANNEL_GROUP_TYPE_ONECALL_DAILY_FORECAST));
+                    }
+                }
+                forecastDays = newForecastDays;
+            }
+            ThingBuilder builder = editThing().withoutChannels(toBeRemovedChannels);
+            for (Channel channel : toBeAddedChannels) {
+                builder.withChannel(channel);
+            }
+            updateThing(builder.build());
```

Comment:
Tried, didn't work. Affected Things are no longer updated then. Reverted to the original code. By the way, the same loop is used in your OpenWeatherMapWeatherAndForecastHandler, maybe for the same reason?

Reply:
IIRC I did not use the suggested code because one of the methods was not available. It has been introduced later. But you are right. It will not work because `withChannels` removes all previously added or available channels and does not work like I expected.